audio-worklet/bitcrusher-test.html

```diff
@@ -15,6 +15,172 @@
+    	        // maximum observed error is 6.183981895446777e-7.
+    	        
+    	        const permittedSampleError = 6.1840e-7;
+    	        let maxError = 0;
+    	        for (let i = 0; i < bitcrusherOutput.length; i++) {
+    	          let diff = Math.abs(bitcrusherOutput[i] - originalOutput);
+    	          if (diff > maxError) maxError = diff;  
+    	        }
+    	        console.assert(
+    	            maxError < permittedSampleError,
+    	            'The maximum error between bitcrusher and original output, ' +
+    	                maxError + ', exceeds the maximum permitted error, ' +
+    	                permittedSampleError);
+
+    	        // Verify samples from unadultered oscillator match samples from
+    	        // bitcrushed oscillator with non-information reducing parameters.
+    	        for (let i = 0; i < originalOutput.length; i++) {
+    	          let crushedSample = bitcrusherOutput[i];
+    	          let originalSample = originalOutput[i];
+    	          const diff = Math.abs(originalSample - crushedSample);
+    	          
+    	          if (i < bufferLength) {
+    	            console.assert(
+    	                bitcrusherOutput[i] == 0,
+    	                'Bitcrusher sample at ' + i + 'expected to be 0 but it'
+    	                + ' was ' + bitcrusherOutput[i]);
+    	          }
+    	          console.assert(
+    	              diff < permittedSampleError,
+    	              'Bitcrushed sample at ' + i + ' is ' + crushedSample + 
+    	              ' but ' + originalSample + ' for oscillator');
+    	        }
+    	      });
+    	}
+
+    	/**
+    	 * Verify that bitcrusher handles illegal input correctly.
+    	 */
+    	function testIllegalInput() {
+    	  try {
+    	    let bitcrusher = new Bitcrusher(0);
+    	    console.error('Bitcrusher did not throw illegal context error.');
+    	  }
+    	  catch (error) {}
+    	}
+
+    	/**
+    	 * Verify that bitcrusher applies sample reduction on multiple channels of 
+    	 * input, and that sample reduction works across multiple buffers (which 
+    	 * requires bitcrusher memory.)
+    	 */
+    	function testMultipleChannels() {
+    	  let channels = 2;
+    	  let context = new OfflineAudioContext(channels, 44100, 44100);
+    	  // Reduction should not be a factor of buffersize for full coverage.
+    	  const reduction = 7; 
+    	  const bitDepth = 24;
+    	  const bufferSize = 512;
+    	  let oscillator = new OscillatorNode(context);
+    	  let splitter = new ChannelSplitterNode(context);
+
+    	  let bitcrusher = new Bitcrusher(context, {
+    	    bufferSize: bufferSize,
+    	    channels: channels,
+    	    bitDepth: bitDepth,
+    	    reduction: reduction
+    	  })
+
+    	  oscillator.connect(splitter).connect(bitcrusher.input);
+    	  bitcrusher.output.connect(context.destination); 
+    	  oscillator.start();
+    	  
+    	  // When audio buffer is ready, verify bitcrushed samples are unaltered.
```

Comment:
In my model of what is going on, when the audio graph has finished rendering it resolves with an audio buffer. But the when clause of this comment is sort of in a narrative style, so I will take it out.